src/citrine/builders/auto_configure.py

```diff
@@ -0,0 +1,183 @@
+    objectives for a scoring function in a design workflow.
+
+    Parameters
+    ----------
+    project: Project
+        Project to use when accessing the Citrine Platform
+    material: Union[MaterialRun, LinkByUID, str, UUID]
+        Sample material to extract descriptor keys
+    mode: str
+        Option in {'SIMPLE', 'FORMULATION'} that controls formatting descriptor keys
+        Default: 'SIMPLE'
+    full_history: bool
+        Whether to extract descriptor keys from the full material history,
+        or only the provided (terminal) material
+        Default: True
+
+    Returns
+    -------
+    dict
+        Dictionary of keys of the form
+        {'properties': [...], 'conditions': [...], 'parameters': []}
+
+    """
+    if mode not in {"FORMULATION", "SIMPLE"}:
+        msg = "Called with mode: {}.  Expected 'SIMPLE' or 'FORMULATION'.".format(mode)
+        raise ValueError(msg)
+
+    link = _make_link_by_uid(material)
+    history = project.material_runs.get_history(scope=link.scope, id=link.id)
+
+    if full_history:
+        search_history = flatten(history)
+    else:
+        # Limit the search to contain the terminal material/process/measurement
+        search_history = [history.spec.template, history.process.template]
+        search_history.extend([msr.template for msr in history.measurements])
+
+    keys = {'properties': [], 'conditions': [], 'parameters': []}
+    for obj in search_history:
+        parent_name = f'{obj.name}~' if mode == 'SIMPLE' else ''
+        # Search all attribute template types
+        if isinstance(obj, HasPropertyTemplates):
+            for property in obj.properties:
+                keys['properties'].append(parent_name + property[0].name)
+        if isinstance(obj, HasConditionTemplates):
+            for condition in obj.conditions:
+                keys['conditions'].append(parent_name + condition[0].name)
+        if isinstance(obj, HasParameterTemplates):
+            for parameter in obj.parameters:
+                keys['parameters'].append(parent_name + parameter[0].name)
+    return keys
+
+
+def auto_configure_material(
+        *,
+        project: Project,
+        material: Union[MaterialRun, LinkByUID, str, UUID],
+        score: Score,
+        mode: str = 'SIMPLE',
+        label: str = '',
+        print_status_info: bool = False,
+) -> DesignExecution:
+    """[ALPHA] Description.
+
+    Parameters
+    ----------
+    project: Project
+        Project to use when accessing the Citrine Platform
+    material: Union[MaterialRun, LinkByUID, str, UUID]
+        Sample material to extract descriptor keys
+    score: Score
+        Scoring function used to rank candidates during design execution.
+        Must contain objectives/constraints with matching descriptor keys
+        to those found within the provided material history.
+    mode: str
+        Option in {'SIMPLE', 'FORMULATION'} that controls formatting descriptor keys
+        Default: 'SIMPLE'
```

Comment:
Is "Simple" the right default?  cc @kroenlein

Reply:
I would guess "SIMPLE" is the expected default -- the most surprising state would be to just pass a Strehlow-like table and get feedback that it doesn't make sense.

Secondary comment: if you are using an enumeration, you should use a BaseEnumeration:
https://github.com/CitrineInformatics/citrine-python/blob/6d8cafc69c511498175fdef77d2195e0ccb49a71/src/citrine/resources/table_config.py#L24

Reply:
@kroenlein Is there ever a case where a user could wish for a `TableBuildAlgorithm.SINGLE_ROW` table, but then a default predictor using the `FORMULATION` option? Or visa-versa?

If not, then I can introduce a new Enum for handling the single option pattern on this method.

Reply:
Linking was explicitly for reference in building your own enumeration, thought reusing isn't a terrible idea.  If a user wanted to reference an existing property table (e.g., someone transcribed something out of Perry's), they might build the main table as a single row (they need the maceration time in the same row as the terminal material) but then want to use a mean property predictor informed by the property table.  But, given that I'm talking about multiple tables as inputs, that might already be deviating outside the scope of this work.